transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollChannel.java

```diff
@@ -61,6 +88,14 @@
         readFlag = flag;
         flags |= flag;
         this.active = active;
+        if (remote != null) {
+            // Directly cache the remote and local addresses
+            // See https://github.com/netty/netty/issues/2359
+            this.remote = remote;
+        }
+        if (active) {
```

Comment:
This will result in a change of behavior in `EpollServerSocketChannel` when a `LinuxSocket` is provided but `active` is `false`. Is this change intentional? Seems reasonable to set local address in bind/connect but I'm not sure if it hurts to set in other scenarios too (and avoid breakage).